lib/matplotlib/figure.py

```diff
@@ -2135,23 +2127,24 @@ def align_xlabels(self, axs=None, renderer=None):
             _log.debug(' Working on: %s', ax.get_xlabel())
             ss = ax.get_subplotspec()
             nrows, ncols, row0, row1, col0, col1 = ss.get_rows_columns()
-            same = [ax]
-            labpo = ax.xaxis.get_label_position()
+            labpo = ax.xaxis.get_label_position() # top or bottom
+
+            # loop through other axes, and search for label positions
+            # that are same as this one, and that share the appropriate
+            # row number.
+            #  Add to a list associated with each axes of sibblings.
+            # This list is inspected in `axis.draw` by
+            # `axis._update_label_position`.
             for axc in axs:
                 if axc.xaxis.get_label_position() == labpo:
                     ss = axc.get_subplotspec()
                     nrows, ncols, rowc0, rowc1, colc, col1 = \
                             ss.get_rows_columns()
-                    if (labpo == 'bottom') and (rowc1 == row1):
-                        same += [axc]
-                    elif (labpo == 'top') and (rowc0 == row0):
-                        same += [axc]
-
-            for axx in same:
-                _log.debug(' Same: %s', axx.xaxis.label)
-                axx.xaxis._align_label_siblings += [ax.xaxis]
+                    if (((labpo == 'bottom') and (rowc1 == row1)) or
```

Comment:
not a big deal but that's a lot of extra parentheses (only the outermost are needed)

Reply:
I guess somewhere in my head I knew that `and` precedes `or`, but I'm never sure, and I'm never sure from language to language, so...